src/chkstat.c

```diff
@@ -441,6 +443,86 @@ usage(int x)
   exit(x);
 }
 
+static bool
+check_have_proc(void)
+{
+  char *override = secure_getenv("CHKSTAT_PRETEND_NO_PROC");
+
+  struct statfs proc;
+  int r = statfs("/proc", &proc);
+  return override == NULL && r == 0 && proc.f_type == PROC_SUPER_MAGIC;
+}
+
+static const char proc_mount_path_pattern[] = "/tmp/chkstat.proc.XXXXXX";
+static char proc_mount_path[sizeof(proc_mount_path_pattern)];
+static int proc_mount_avail = 0;
```

Comment:
Could we please have an `enum` for this?

Reply:
Done